zanata-war/src/main/resources/db/changelogs/db.changelog-3.6.xml

```diff
@@ -30,4 +30,116 @@
+    <sql>ALTER TABLE HDocument_RawDocument ENGINE = INNODB</sql>
+  </changeSet>
+  <changeSet id="3.10" author="sflaniga@redhat.com" dbms="mysql">
+    <comment>Ensure InnoDB for HDocumentUpload</comment>
+    <sql>ALTER TABLE HDocumentUpload ENGINE = INNODB</sql>
+  </changeSet>
+  <changeSet id="3.11" author="sflaniga@redhat.com" dbms="mysql">
+    <comment>Ensure InnoDB for HDocumentUploadPart</comment>
+    <sql>ALTER TABLE HDocumentUploadPart ENGINE = INNODB</sql>
+  </changeSet>
+  <changeSet id="3.12" author="sflaniga@redhat.com" dbms="mysql">
+    <comment>Ensure InnoDB for HAccountOption</comment>
+    <sql>ALTER TABLE HAccountOption ENGINE = INNODB</sql>
+  </changeSet>
+  <changeSet id="3.13" author="sflaniga@redhat.com" dbms="mysql">
+    <comment>Ensure InnoDB for HPersonEmailValidationKey</comment>
+    <sql>ALTER TABLE HPersonEmailValidationKey ENGINE = INNODB</sql>
+  </changeSet>
+  <changeSet id="3.14" author="sflaniga@redhat.com" dbms="mysql">
+    <comment>Ensure InnoDB for HProject_Validation</comment>
+    <sql>ALTER TABLE HProject_Validation ENGINE = INNODB</sql>
+  </changeSet>
+  <changeSet id="3.15" author="sflaniga@redhat.com" dbms="mysql">
+    <comment>Ensure InnoDB for HProjectIteration_Validation</comment>
+    <sql>ALTER TABLE HProjectIteration_Validation ENGINE = INNODB</sql>
+  </changeSet>
+  <changeSet id="3.16" author="sflaniga@redhat.com" dbms="mysql">
+    <comment>Ensure InnoDB for TransMemory</comment>
+    <sql>ALTER TABLE TransMemory ENGINE = INNODB</sql>
+  </changeSet>
+  <changeSet id="3.17" author="sflaniga@redhat.com" dbms="mysql">
+    <comment>Ensure InnoDB for TransMemory_Metadata</comment>
+    <sql>ALTER TABLE TransMemory_Metadata ENGINE = INNODB</sql>
+  </changeSet>
+  <changeSet id="3.18" author="sflaniga@redhat.com" dbms="mysql">
+    <comment>Ensure InnoDB for TransMemoryUnit</comment>
+    <sql>ALTER TABLE TransMemoryUnit ENGINE = INNODB</sql>
+  </changeSet>
+  <changeSet id="3.19" author="sflaniga@redhat.com" dbms="mysql">
+    <comment>Ensure InnoDB for TransMemoryUnitVariant</comment>
+    <sql>ALTER TABLE TransMemoryUnitVariant ENGINE = INNODB</sql>
+  </changeSet>
+  <changeSet id="3.20" author="sflaniga@redhat.com" dbms="mysql">
+    <comment>Ensure InnoDB for HTextFlowTargetReviewComment</comment>
+    <sql>ALTER TABLE HTextFlowTargetReviewComment ENGINE = INNODB</sql>
+  </changeSet>
+  <!-- changeset 3.21: HRawDocumentContent no longer exists -->
+  <changeSet id="3.22" author="sflaniga@redhat.com" dbms="mysql">
+    <!-- Activity table can be big, so check before running -->
+    <preConditions onFail="MARK_RAN">
+      <sqlCheck expectedResult="0">
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.TABLES
+        WHERE TABLE_SCHEMA = SCHEMA()
+        AND ENGINE = 'InnoDB'
+        AND TABLE_NAME='Activity';
+      </sqlCheck>
+    </preConditions>
+    <comment>Ensure InnoDB for Activity</comment>
+    <sql>ALTER TABLE Activity ENGINE = INNODB</sql>
```

Comment:
Well,

    <!-- Activity table can be big, so check before running -->

None of the other tables was big enough to concern me, but I thought it was *possible* that this conversion might take a while, so the precondition lets us skip it if we don't need it.  

Only on some environments have we been creating MyISAM tables; others are using InnoDB everywhere.  I suspect it is MySQL 5.1 that doesn't (always?) respect the session variable.